yorkie/src/androidTest/kotlin/dev/yorkie/core/ClientTest.kt

```diff
@@ -147,13 +154,17 @@ class ClientTest {
             assertEquals(1, document2.clone?.getGarbageLength())
 
             client1.updatePresenceAsync("k2", "v2").await()
-            peerStatus = client2.peerStatus.dropWhile { it == peerStatus }.first()
-            val status = peerStatus.first { it.actorId == client1.requireClientId() }
-            assertEquals(mapOf("k2" to "v2"), status.presenceInfo.data)
+            peerStatus = client2.peerStatusByDoc(documentKey)
+                .dropWhile {
+                    it == peerStatus
+                }.drop(1).first()
+
+            val status = peerStatus.entries.first { it.key == client1.requireClientId() }
+            assertEquals(mapOf("k2" to "v2"), status.value.data)
             assertTrue(
-                peerStatus.first {
-                    it.actorId == client2.requireClientId()
-                }.presenceInfo.data.isEmpty(),
+                peerStatus.entries.first {
+                    it.key == client2.requireClientId()
+                }.value.data.isEmpty(),
```

Comment:
it still seems to cause failures intermittently, but not sure why🤔

Reply:
there were two issues :(
Peers class' equality check was only comparing references and not its contents, so `dropWhile` was not working as my intention.
Also, I changed presence property in Attachment to immutable, and always emitted new Attachments when there were any changes, which caused `drop(1)` to be unstable